### PR TITLE
agent: Add start/stop resource to rust agent

### PR DESCRIPTION
**Pull request by utopiabound**
This is to suppliment python agent `start_target`.  It does not do any movement if the resources is started, nor does it try to force a start in a particular place.

Related #1464 

Signed-off-by: Nathaniel Clark <nclark@whamcloud.com>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2146)

### CODE REVIEW

File: iml-agent/src/action_plugins/action_plugin.rs [hunk 1]
```diff
@@ -45,6 +45,8 @@ pub fn create_registry() -> action_plugins::Actions {
         .add_plugin("mount_many", lustre::client::mount_many)
         .add_plugin("unmount", lustre::client::unmount)
         .add_plugin("unmount_many", lustre::client::unmount_many)
+        .add_plugin("ha_resource_start", high_availability::start_resource)
+        .add_plugin("ha_resource_stop", high_availability::stop_resource)
```
Comment (ip1981) [inline iml-agent/src/action_plugins/action_plugin.rs:49]:
```suggestion
        .add_plugin("ha_start_resource", high_availability::start_resource)
        .add_plugin("ha_stop_resource", high_availability::stop_resource)
```
Just for consistence :)

**Comment by jgrund** [comment]
Why does this supplement the python `start_target` instead of replacing it?

**Comment by utopiabound** [comment]
> Why does this supplement the python `start_target` instead of replacing it?

Because of all the other work that `start_target` does (looking for paired ZFS resources, and movement of resources)

File: iml-agent/src/cli.rs [hunk 2]
```diff
@@ -552,24 +567,38 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
...
+                }
+                Err(e) => eprintln!("{}", e),
+            },
+            HighAvailability::List => match high_availability::get_ha_resource_list(()).await {
+                Ok(v) => {
+                    for e in v {
+                        println!("{}", serde_json::to_string(&e).unwrap())
+                    }
+                }
+                Err(e) => eprintln!("{:?}", e),
```
Comment (ip1981) [inline iml-agent/src/cli.rs:588]: Isn't exit needed here?

File: iml-agent/src/cli.rs [hunk 3]
```diff
@@ -552,24 +567,38 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
...
+            HighAvailability::Check => match high_availability::check_ha(()).await {
+                Ok((cs, pm, pc)) => {
+                    let mut table = Table::new();
+                    table.add_row(row!["Name", "Config", "Service"]);
+                    table.add_row(row!["corosync", cs.config, cs.service]);
+                    table.add_row(row!["pacemaker", pm.config, pm.service]);
+                    table.add_row(row!["pcsd", pc.config, pc.service]);
+                    table.printstd();
+                }
+                Err(e) => eprintln!("{}", e),
```
Comment (ip1981) [inline iml-agent/src/cli.rs:580]: Below it is mostly `{:?}`

File: iml-agent/src/action_plugins/high_availability/mod.rs [hunk 4]
```diff
@@ -61,6 +63,105 @@ where
...
+async fn crm_mon_status() -> Result<Element, ImlAgentError> {
+    let o = Command::new("crm_mon")
+        .args(&["--as-xml", "--inactive"])
+        .checked_output()
+        .await?;
+
+    Ok(Element::from_reader(o.stdout.as_slice())?)
+}
+
+fn resource_status(tree: Element) -> Result<HashMap<String, String>, ImlAgentError> {
```
Comment (jgrund) [inline iml-agent/src/action_plugins/high_availability/mod.rs:81]: Why does this need to return a result?

File: iml-agent/src/action_plugins/high_availability/mod.rs [hunk 5]
```diff
@@ -61,6 +63,105 @@ where
...
+                    }
+                })
+                .flatten()
+        })
+        .flatten()
+        .map(|(x, y)| (x.to_string(), y.to_string()))
+        .collect())
+}
+
+async fn set_resource_role(resource: &str, running: bool) -> Result<(), ImlAgentError> {
```
Comment (jgrund) [inline iml-agent/src/action_plugins/high_availability/mod.rs:121]: Why does this need to return a result?

Comment (ip1981) [reply iml-agent/src/action_plugins/high_availability/mod.rs:121]: `crm_resource(args).await?;` below?

Comment (jgrund) [reply iml-agent/src/action_plugins/high_availability/mod.rs:121]: Yup, missed that.

**Comment by johnsonw** [comment]
Looks like there is a failure in the integration tests.

File: iml-agent/src/action_plugins/high_availability/mod.rs [hunk 6]
```diff
@@ -85,6 +87,105 @@ where
...
+    let o = Command::new("crm_mon")
+        .args(&["--as-xml", "--inactive"])
+        .checked_output()
+        .await?;
+
+    Ok(Element::from_reader(o.stdout.as_slice())?)
+}
+
+fn resource_status(tree: Element) -> HashMap<String, String> {
+    let resources = tree.find("resources").unwrap();
```
Comment (jgrund) [inline iml-agent/src/action_plugins/high_availability/mod.rs:106]: Nit: this is similar to what I'm doing in terms of reading crm_mon resources. We should consolidate our changes.

**Pull request merged**